libs/database/src/lib/entities/project-report.entity.ts

```diff
@@ -96,6 +96,15 @@ export class ProjectReport extends Model<ProjectReport> {
     return builder.literal;
   }
 
+  static approvedIdsSubqueryProjects(projectIds: number[], opts: ApprovedIdsSubqueryOptions = {}) {
+    const builder = Subquery.select(ProjectReport, "id")
+      .in("projectId", projectIds)
+      .in("status", ProjectReport.APPROVED_STATUSES);
+    if (opts.dueAfter != null) builder.gte("dueAt", opts.dueAfter);
+    if (opts.dueBefore != null) builder.lt("dueAt", opts.dueBefore);
```

Comment:
It looks like nothing is using this due after / before option currently. Unless you know you're going to need it, let's not add it for now.